src/parser.cpp

```diff
@@ -498,11 +501,11 @@ void Parser::ParserImpl::loadUnit(const UnitsPtr &units, const XmlNodePtr &node)
                 err->setKind(Error::Kind::UNITS);
                 err->setRule(SpecificationRule::UNIT_MULTIPLIER);
                 mParser->addError(err);
-            } else {
-                multiplier = std::stod(attribute->getValue());
             }
         } else if (attribute->isType("offset")) {
-            if (mParser->catchDoubleConversionError(attribute->getValue())) {
+            if (canConvertToDouble(attribute->getValue())) {
+                offset = std::stod(attribute->getValue());
+            } else {
```

Comment:
If I'm right you're saying that we should convert to double and return the success failure by return value.